lib/Auth/Process/UpdateAUP.php

```diff
@@ -79,7 +79,7 @@ public function process(&$state)
             $changed_aups = array();
 
             foreach ($state['rciamAttributes']['aup'] as $aup) {
-                if (!empty($aup['agreed']) && $aup['version'] != $aup['agreed']['version']) {
+                if (empty($aup['agreed']) || (!empty($aup['agreed']) && $aup['version'] != $aup['agreed']['version'])) {
```

Comment:
isn't this equivalent to
`if (empty($aup['agreed']) || $aup['version'] != $aup['agreed']['version'])` ?

Reply:
yes you are right! i definetely need a break :P